Aplib.Core/Belief/Belief.cs

```diff
@@ -30,57 +30,88 @@ public class Belief<TReference, TObservation> : IBelief where TReference : class
-        /// reference type, be aware that this is not enforced by C# if <typeparamref name="TReference"/> is an
-        /// interface.
+        /// The object reference used to generate/update the observation. This <i>must</i> be a reference type, be aware
+        /// that this is not enforced by C# if <typeparamref name="TReference"/> is an interface.
+        /// </param>
+        /// <param name="getObservationFromReference">
+        /// A function that takes an object reference and generates/updates an observation.
         /// </param>
-        /// <param name="getObservationFromReference">A function that takes an object reference and generates/updates an observation.</param>
+        /// <param name="shouldUpdate">A condition on when the observation should be updated.</param>
         /// <exception cref="ArgumentException">
         /// Thrown when <paramref name="reference"/> is not a reference type.
         /// </exception>
-        public Belief(TReference reference, Func<TReference, TObservation> getObservationFromReference)
+        public Belief
+        (
+            Metadata metadata,
+            TReference reference,
+            Func<TReference, TObservation> getObservationFromReference,
+            Func<bool> shouldUpdate
+        )
         {
             Type referenceType = reference.GetType();
             if (referenceType.IsValueType)
                 throw new ArgumentException($"{referenceType.FullName} is not a reference type.", nameof(reference));
 
+            Metadata = metadata;
             _reference = reference;
             _getObservationFromReference = getObservationFromReference;
             Observation = _getObservationFromReference(_reference);
+            _shouldUpdate = shouldUpdate;
         }
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="Belief{TReference, TObservation}"/> class with an object reference,
-        /// a function to generate/update the observation using the object reference,
-        /// and a condition on when the observation should be updated.
-        /// </summary>
-        /// <param name="reference">
-        /// The object reference used to generate/update the observation. This <i>must</i> be a reference type, be aware
-        /// that this is not enforced by C# if <typeparamref name="TReference"/> is an interface.
-        /// </param>
-        /// <param name="getObservationFromReference">A function that takes an object reference and generates/updates an observation.</param>
-        /// <param name="shouldUpdate">A condition on when the observation should be updated.</param>
-        /// <exception cref="ArgumentException">
-        /// Thrown when <paramref name="reference"/> is not a reference type.
-        /// </exception>
-        public Belief(TReference reference,
+        /// <inheritdoc
+        ///     cref="Belief{TReference,TObservation}(Aplib.Core.Metadata,TReference,Func{TReference,TObservation},Func{bool})"/>
```

Comment:
That is the magic of `inheritdoc`

Reply:
This means we don't have to write every param again, removing docs duplication.

Reply:
Rider shows it anyway if I'm not mistaken, but that's a bug in Rider.

Reply:
That's indeed why I mentioned it, Rider. And then again, not so much magic if you have to exhaustively mention everything. What if a field is added? Then, we need to manually expand the inherit doc. Isn't it possible to just inherit all?

Reply:
Right now it is inheriting all, if I understand you correctly. Inheritdoc takes all applicable XML tags on the referenced member. We don't do this to prevent work for ourselves in the short run, but to prevent missed docs changes if we ever want to change the XML docs on a member. With inheritdoc, we only need to change the docs in one place, and all other overloads would then also use the updated XML docs.

Reply:
Having to say `inheritdoc cref="fieldA, fieldB, fieldC" is not the same as including 'all', just explicitly every current field. If field D is added someday, it will not be included. I will resolve this to move on